### PR TITLE
Don't enable IvyPlugin on artificial root of a multi project build

### LINKED ISSUE #1869: artificial root should not publish by default
Suggested by @softprops at nescala. Please comment on this issue, if someone wants to send PR for this.

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #1869, Ref #1871

/review @jsuereth

### CODE REVIEW

**Comment by jsuereth** [comment]
one comment, otherwise looks like it might work

File: main/src/main/scala/sbt/Project.scala [hunk 1]
```diff
@@ -234,6 +234,13 @@ object Project extends ProjectExtra {
     auto: AddSettings = AddSettings.allDefaults): Project =
     unresolved(id, base, aggregate, dependencies, delegates, settings, configurations, auto, Plugins.empty, Nil) // Note: JvmModule/IvyModule auto included...
 
+  /** This is a variation of def apply that mixes in  */
```
Comment (dwijnand) [inline main/src/main/scala/sbt/Project.scala]: Looks like this sentence needs finishing.

File: sbt/src/sbt-test/project/artificial-root-no-publish/test [hunk 2]
```diff
@@ -0,0 +1,10 @@
+> publishLocal
+$ exists target/ivy-cache/local/com.example/app_2.10/0.1.0/poms/app_2.10.pom
+$ exists target/ivy-cache/local/com.example/app_2.10/0.1.0/jars/app_2.10.jar
+$ absent target/ivy-cache/local/com.example/default-root_2.10/0.1.0/ivys/ivy.xml
+
+$ copy-file changes/bare.sbt build.sbt
+> reload
+> publishLocal
+$ exists target/ivy-cache/local/com.example/artificial-root-no-publish_2.10/0.1.0/poms/artificial-root-no-publish_2.10.pom
+$ exists target/ivy-cache/local/com.example/artificial-root-no-publish_2.10/0.1.0/jars/artificial-root-no-publish_2.10.jar
```
Comment (dwijnand) [inline sbt/src/sbt-test/project/artificial-root-no-publish/test]: This could be named `generated-root-no-publish` so everything matches..

**Comment by eed3si9n** [comment]
The Travis status looks confused, but all tests have passed - https://travis-ci.org/sbt/sbt/builds/52091079

File: main/src/main/scala/sbt/Load.scala [hunk 3]
```diff
@@ -585,7 +585,9 @@ object Load {
               val existingIds = otherProjects.projects map (_.id)
               val refs = existingIds map (id => ProjectRef(buildUri, id))
               val defaultID = autoID(buildBase, context, existingIds)
-              val root = finalizeProject(Build.defaultAggregatedProject(defaultID, buildBase, refs), files)
+              val root0 = if (discovered.isEmpty || java.lang.Boolean.getBoolean("sbt.root.ivyplugin")) Build.defaultAggregatedProject(defaultID, buildBase, refs)
```
Comment (jsuereth) [inline main/src/main/scala/sbt/Load.scala:588]: This doesn't look right. getBoolean on the raw string instead of the actual property variable....

**Comment by eed3si9n** [comment]
Merged. @dwijnand Thanks for your patience.

**Comment by dwijnand** [comment]
Wohoo, thanks.

**Pull request merged**